### PR TITLE
Feature/generic improvements

### PR DESCRIPTION
**Pull request by lrascao**
This first PR is a sort of getting acquainted with the project and first getting it to work on my regular use-case.
Mostly cosmetic except 6b1212c and 0825fe8 which were actually stopping from successfully building.
Other cosmetic changes are open to discussion as personal tastes might vary ofc.

### CODE REVIEW

**Comment by lrascao** [comment]
here's a screenshot of a sample run:
https://imgur.com/a/zNDeeUM

**Comment by tsloughter** [comment]
Looks great. I just wanted to verify, did you test with a user options like in the doc, `user_configure_options: ["--without-wx", "--without-odbc", "CFLAGS=-g -O2 -march=native"]`? I'm just concerned that the `CFLAGS` without quoting `-g -O2 -march=native` might not work.

**Review by tsloughter — APPROVED** [review]

**Comment by lrascao** [comment]
> Looks great. I just wanted to verify, did you test with a user options like in the doc, `user_configure_options: ["--without-wx", "--without-odbc", "CFLAGS=-g -O2 -march=native"]`? I'm just concerned that the `CFLAGS` without quoting `-g -O2 -march=native` might not work.

i'll double check it

**Comment by tsloughter** [comment]
Oh, looks like it must work, your screenshot shows it using the `CFLAGS` arg.

**Comment by lrascao** [comment]
actually without the quotes it won't work

this works:
`default_configure_options=--without-wx --without-observer --without-odbc --without-debugger --without-et --enable-builtin-zlib --without-javac CFLAGS="-g -O2 -march=native"`

without the quotes it doesn't:
`default_configure_options=--without-wx --without-observer --without-odbc --without-debugger --without-et --enable-builtin-zlib --without-javac CFLAGS=-g -O2 -march=native`

but you need the quotes right? how will configure know what `CFLAGS` is?

**Comment by tsloughter** [comment]
@lrascao oh, not what I meant. I should have linked to the spot in the code, one sec :)

File: src/build.rs [hunk 1]
```diff
@@ -330,49 +340,83 @@ pub fn build(
...
-                ("make", &["docs", "DOC_TARGETS=chunks"]),
-                ("make", &["install"]),
-                ("make", &["install-docs"]),
+
+            // split the configure options into a vector of String in a shell sensitive way
+            // eg.
+            //  from:
+            //      user_configure_options0: --without-wx --without-odbc CFLAGS=\"-g -O2 -march=native\"
+            //  to:
+            //      user_configure_options: ["--without-wx", "--without-odbc", "CFLAGS=-g -O2 -march=native"]
```
Comment (tsloughter) [inline src/build.rs]: I wanted to make sure that this didn't need to be `"CFLAGS=\"-g -O2 -march=native\""`

Comment (tsloughter) [reply src/build.rs]: Maybe it already is actually keeping the quotes?

Comment (lrascao) [reply src/build.rs]: ah no, at least not in `default_configure_options` (as shown above), not so sure on the command line now...

Comment (lrascao) [reply src/build.rs]: comment is wrong, should be:
```
            // split the configure options into a vector of String in a shell sensitive way
            // eg.
            //  from:
            //      user_configure_options0: --without-wx --without-observer --without-odbc --without-debugger --without-et --enable-builtin-zlib --without-javac CFLAGS="-g -O2 -march=native"
            //  to:
            //      user_configure_options: ["--without-wx", "--without-observer", "--without-odbc", "--without-debugger", "--without-et", "--enable-builtin-zlib", "--without-javac", "CFLAGS=-g -O2 -march=native"]

```

Comment (lrascao) [reply src/build.rs]: i'm updating the PR, also found that if configure fails we don't error out

Comment (lrascao) [reply src/build.rs]: updated now

Comment (tsloughter) [reply src/build.rs]: Ok, great.

**Pull request merged**